--- a/output-translations.lisp
+++ b/output-translations.lisp
@@ -146,6 +146,7 @@
     '(environment-output-translations
       user-output-translations-pathname
       user-output-translations-directory-pathname
+      nix-profile-output-translations
       system-output-translations-pathname
       system-output-translations-directory-pathname))
 
@@ -184,6 +185,14 @@
                          :direction direction))
   (defun environment-output-translations ()
     (getenv "ASDF_OUTPUT_TRANSLATIONS"))
+  (defun nix-profile-output-translations ()
+    `(:output-translations
+      ,@(loop :for profile :in (reverse (split-string (or (getenv "NIX_PROFILES") "")))
+              :for share := (subpathname (ensure-directory-pathname profile) "share/")
+              :for sysdir := (namestring (subpathname share "common-lisp/systems/"))
+              :for fasldir := (namestring (subpathname share "common-lisp/fasl/"))
+              :collect `(,sysdir (,fasldir :implementation)))
+      :inherit-configuration))
 
 
   ;;; Processing the configuration.

--- a/source-registry.lisp
+++ b/source-registry.lisp
@@ -187,6 +187,7 @@
       user-source-registry
       user-source-registry-directory
       default-user-source-registry
+      nix-profile-source-registry
       system-source-registry
       system-source-registry-directory
       default-system-source-registry)
@@ -227,5 +228,11 @@
                          :direction direction))
   (defun environment-source-registry ()
     (getenv "CL_SOURCE_REGISTRY"))
+  (defun nix-profile-source-registry ()
+    `(:source-registry
+      ,@(loop :for profile :in (reverse (split-string (or (getenv "NIX_PROFILES") "")))
+              :for share := (subpathname (ensure-directory-pathname profile) "share/")
+              :collect `(:tree ,(namestring (subpathname share (parse-unix-namestring "common-lisp/systems/")))))
+      :inherit-configuration))
 
 
   ;;; Process the source-registry configuration