--- a/babel.asd
+++ a/babel.asd
@@ -52,6 +52,14 @@
      (:file "gbk-map")
      (:file "sharp-backslash")))))
 
+(defsystem babel/foo
+  :description "Babel, a charset conversion library."
+  :author "Luis Oliveira <loliveira@common-lisp.net>"
+  :licence "MIT"
+  :depends-on ("babel")
+  :components
+  ((:file "bla")))
+
 (defmethod perform ((o test-op) (c (eql (find-system :babel))))
   (operate 'load-op :babel-tests)
   (operate 'test-op :babel-tests))

Diff finished.  Sun Jun 16 11:21:26 2024
